phpunit/style-engine/class-wp-style-engine-gutenberg-test.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Tests the Style Engine class and associated functionality.
+ *
+ * @package    Gutenberg
+ * @subpackage block-library
+ */
+
+/**
+ * Tests for registering, storing and generating styles.
+ */
+class WP_Style_Engine_Gutenberg_Test extends WP_UnitTestCase {
+	function test_returns_inline_styles_from_string() {
+		$style_engine   = WP_Style_Engine_Gutenberg::get_instance();
```

Comment:
Does it need to be a class? Why not just a function, mostly curious about the reasoning here.

Reply:
Good question. It doesn't have to be a class. The reasons why I left it as a class are:

- because I copied the structure over from one of our initial experiments 😄 
- to namespace methods and encapsulate constants such as `BLOCK_STYLE_DEFINITIONS_METADATA` (similar to  `WP_Theme_JSON_Gutenberg`)
- portability across applications
- related to the last point, so we can be free to use generic method names such as `generate` to harmonize the backend and frontend APIs, as mentioned [below](https://github.com/WordPress/gutenberg/pull/39446#discussion_r832150344).
- while a more conceptual point, to lend the functionality a degree of authority and conspicuousness, that is, so folks recognize it as a unified collection of properties and behaviours.

Reply:
Good question, and good answers! 😁

> while a more conceptual point, to lend the functionality a degree of authority and conspicuousness, that is, so folks recognize it as a unified collection of properties and behaviours.

Yes, in PHP, because we're not using namespaces and can't encapsulate our functions in a separate module like we do in JS, I think it's better for us to gather together our style engine functions in a class rather than expose each of them in the global namespace. Plus, it's looking like we'll likely need to carry around some state in the style engine to deal with de-duping styles and / or deferring processing to a later part of the execution process, so starting with a class likely makes it easier for us to extend the style engine in follow-up PRs, too.

Reply:
Makes sense. I just wanted us to avoid the "stateful" trap if possible. When we add state to classes that are not meant to be classes, the problems start happening :)